lib/router.dart

```diff
@@ -13,11 +16,25 @@ import 'package:flex_storefront/root/root_page.dart';
 import 'package:flex_storefront/search/models/search_results.dart';
 import 'package:flex_storefront/shop/shop_page.dart';
 import 'package:flutter/material.dart';
+import 'package:get_it/get_it.dart';
 import 'package:sheet/route.dart';
 import 'package:sheet/sheet.dart';
 
 part 'router.gr.dart';
 
+class AnalyticsNavigationObserver extends AutoRouterObserver {
+  @override
+  void didChangeTabRoute(TabPageRoute route, TabPageRoute previousRoute) {
+    if (route.name == CartRoute.name) {
+      final cart = GetIt.instance.get<CartRepository>().hasCart
+          ? GetIt.instance.get<CartRepository>().latestCart
+          : null;
+
+      GetIt.instance.get<AnalyticsApi>().track(ViewCartEvent.fromCart(cart));
```

Comment:
So yes, the "ViewCartEvent" can be fired with no data attached to it... but that's the best I could come up with.

Either we fire the event at the navigation level so we are sure to fire the "ViewCartEvent" when the user "views the cart", or we fire it at the Bloc level where we are sure about the data, but it no longer correspond to the "View cart" action.